pkg/flow/streaming/flow_topn.go

```diff
@@ -0,0 +1,171 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package streaming
+
+import (
+	"github.com/emirpasic/gods/maps/treemap"
+	"github.com/emirpasic/gods/utils"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+
+	modelv1 "github.com/apache/skywalking-banyandb/api/proto/banyandb/model/v1"
+	"github.com/apache/skywalking-banyandb/pkg/flow"
+)
+
+type windowedFlow struct {
+	f  *streamingFlow
+	wa flow.WindowAssigner
+}
+
+func (s *windowedFlow) TopN(topNum int, opts ...any) flow.Flow {
+	s.wa.(*TumblingTimeWindows).aggregationFactory = func() flow.AggregationOp {
+		topNAggrFunc := &topNAggregator{
+			cacheSize: topNum,
+			sort:      modelv1.Sort_SORT_UNSPECIFIED,
+		}
+		// apply user customized options
+		for _, opt := range opts {
+			if applier, ok := opt.(TopNOption); ok {
+				applier(topNAggrFunc)
+			}
+		}
+		if topNAggrFunc.sortKeyExtractor == nil {
+			s.f.drainErr(errors.New("sortKeyExtractor must be specified"))
+		}
+		if topNAggrFunc.sort == modelv1.Sort_SORT_DESC {
+			topNAggrFunc.comparator = func(a, b interface{}) int {
+				return utils.Int64Comparator(b, a)
+			}
+		} else {
+			topNAggrFunc.comparator = utils.Int64Comparator
+		}
```

Comment:
The order should default to DESC if users don't specific one.

Reply:
Fixed. And according to our last discussion, an Enum is introduced in `streaming` package.